### PR TITLE
create models

### PR DESCRIPTION
**Pull request by NguyenCongViet**
ready

### CODE REVIEW

File: app/Models/Image.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * [$fillable description]
+     * @var [type]
+     */
+    protected $fillable = [
+        'post_id',
+        'filename',
+        'created_at',
+        'updated_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Image.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Image.php</br>Comma required after last value in array declaration

File: app/Models/Image.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Get posts: One to many
+     * @return [type] [description]
+     */
+    public function posts() {
+        return $this->belongsTo('App\Models\Post', 'post_id');
```
Comment (framgiaciv3) [inline app/Models/Image.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Image.php</br>Opening brace should be on a new line

File: app/Models/Role.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+class Role extends Model
+{
+    use SoftDeletes;
+    /**
+     * [$table description]
+     * @var string
+     */
+    protected $table="roles";
+    /**
```
Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Models/Role.php</br>String "roles" does not require double quotes; use single quotes instead

File: app/Models/Role.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        'name',
+        'display_name',
+        'description',
+    ];
+    /**
+     * Tạo liên kết với bảng role_user qua role_id
+     * @return [type] [description]
+     */
+    public function users(){
+        return $this->belongsToMany('App\Models\User','role_user','role_id', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Role.php</br>Opening brace should be on a new line

File: app/Models/Role.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
```
Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Models/Role.php</br>No space found after comma in function call

Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Models/Role.php</br>No space found after comma in function call

File: app/Models/Role.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Tạo liên kết với bảng permission_user qua role_id
+     * @return [type] [description]
+     */
+    public function permissions(){
+        return $this->belongsToMany('App\Models\Permission','permission_role','role_id', 'permission_id');
```
Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Role.php</br>Opening brace should be on a new line

File: app/Models/Role.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
```
Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Models/Role.php</br>No space found after comma in function call

Comment (framgiaciv3) [inline app/Models/Role.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Models/Role.php</br>No space found after comma in function call

File: app/Models/RoleUser.php [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+class RoleUser extends Model
+{
+    use SoftDeletes;
+    /**
+     * [$table description]
+     * @var string
+     */
+    protected $table="role_user";
+    /**
```
Comment (framgiaciv3) [inline app/Models/RoleUser.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Models/RoleUser.php</br>String "role_user" does not require double quotes; use single quotes instead

File: app/Models/Post.php [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        'user_id',
+        'topic_id',
+        'title',
+        'slug',
+        'body',
+        'type',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Comma required after last value in array declaration

File: app/Models/Post.php [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * Get images: One to many
+     * @return [type] [description]
+     */
+    public function images() {
+        return $this->hasMany('App\Models\Image', 'post_id');
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Opening brace should be on a new line

File: app/Models/Post.php [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    /**
+     * Get reports: One to many
+     * @return [type] [description]
+     */
+    public function reports() {
+        return $this->hasMany('App\Models\Report', 'post_id');
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Opening brace should be on a new line

File: app/Models/Post.php [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    /**
+     * Get comments: One to many
+     * @return [type] [description]
+     */
+    public function comments() {
+        return $this->hasMany('App\Models\Comment', 'post_id');
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Opening brace should be on a new line

File: app/Models/Post.php [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    /**
+     * Get topics: One to many
+     * @return [type] [description]
+     */
+    public function topics() {
+        return $this->belongsTo('App\Models\Topic', 'post_id');
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Opening brace should be on a new line

File: app/Models/Post.php [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    /**
+     * Get users: One to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Post.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Post.php</br>Opening brace should be on a new line

File: app/Models/Like.php [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+     * @var [type]
+     */
+    protected $fillable = [
+        'user_id',
+        'post_id',
+        'type',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Like.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Like.php</br>Comma required after last value in array declaration

File: app/Models/Like.php [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Get users: One to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Like.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Like.php</br>Opening brace should be on a new line

File: app/Models/Report.php [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+     * @var [type]
+     */
+    protected $fillable = [
+        'user_id',
+        'post_id',
+        'type',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Report.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Report.php</br>Comma required after last value in array declaration

File: app/Models/Report.php [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Get users: One to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Report.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Report.php</br>Opening brace should be on a new line

File: app/Models/User.php [hunk 19]
```diff
@@ -31,4 +44,39 @@ class User extends Authenticatable
         'password',
         'remember_token',
     ];
+    /**
+     * Get topics: One to many
```
Comment (framgiaciv3) [inline app/Models/User.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/User.php</br>Comma required after last value in array declaration

File: app/Models/User.php [hunk 20]
```diff
@@ -31,4 +44,39 @@ class User extends Authenticatable
...
+     * Get posts: One to many
+     * @return [type] [description]
+     */
+    public function posts() {
+        return $this->hasMany('App\Models\Post', 'user_id');
+    }
+    /**
+     * Get comments: One to many
+     * @return [type] [description]
+     */
```
Comment (framgiaciv3) [inline app/Models/User.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/User.php</br>Opening brace should be on a new line

File: app/Models/User.php [hunk 21]
```diff
@@ -31,4 +44,39 @@ class User extends Authenticatable
...
+    public function comments() {
+        return $this->hasMany('App\Models\Comment', 'user_id');
+    }
+    /**
+     * Get confessions: Many to many
+     * @return [type] [description]
+     */
```
Comment (framgiaciv3) [inline app/Models/User.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/User.php</br>Opening brace should be on a new line

File: app/Models/User.php [hunk 22]
```diff
@@ -31,4 +44,39 @@ class User extends Authenticatable
...
+    public function confessions() {
+        return $this->belongsToMany('App\Models\Confession', 'confession_user', 'user_id', 'confession_id');
+    }
+    /**
+     * Get all of the user's follows.
+     */
+    public function follows()
```
Comment (framgiaciv3) [inline app/Models/User.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/User.php</br>Opening brace should be on a new line

File: app/Models/Topic.php [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+        'name',
+        'slug',
+        'images',
+        'set_time',
+        'select_time',
+        'status',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Topic.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Topic.php</br>Comma required after last value in array declaration

File: app/Models/Topic.php [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Get posts: One to many
+     * @return [type] [description]
+     */
+    public function posts() {
+        return $this->hasMany('App\Models\Post', 'topic_id');
```
Comment (framgiaciv3) [inline app/Models/Topic.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Topic.php</br>Opening brace should be on a new line

File: app/Models/Topic.php [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    /**
+     * Get users: Many to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Topic.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Topic.php</br>Opening brace should be on a new line

File: app/Models/Permission.php [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+
+class Permission extends Model
+{
+    use SoftDeletes;
+    /**
+     * [$table description]
+     * @var string
+     */
+    protected $table="permissions";
+    /**
```
Comment (framgiaciv3) [inline app/Models/Permission.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Models/Permission.php</br>String "permissions" does not require double quotes; use single quotes instead

File: app/Models/Permission.php [hunk 27]
```diff
@@ -0,0 +1,37 @@
...
+        'name',
+        'display_name',
+        'description',
+    ];
+    /**
+     * Get roles
+     * @return [type] [description]
+     */
+    public function roles() {
+        return $this->belongsToMany('App\Models\Role', 'permission_role', 'permission_id', 'role_id');
```
Comment (framgiaciv3) [inline app/Models/Permission.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Permission.php</br>Opening brace should be on a new line

File: app/Models/PermissionRole.php [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+
+class PermissionRole extends Model
+{
+    use SoftDeletes;
+    /**
+     * [$table description]
+     * @var string
+     */
+    protected $table="permission_role";
+    /**
```
Comment (framgiaciv3) [inline app/Models/PermissionRole.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Models/PermissionRole.php</br>String "permission_role" does not require double quotes; use single quotes instead

File: app/Models/Follow.php [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    protected $fillable = [
+        'user_id',
+        'follow_id',
+        'follow_type',
+        'type',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Follow.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Follow.php</br>Comma required after last value in array declaration

File: app/Models/Follow.php [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Get users: One to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Follow.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Follow.php</br>Opening brace should be on a new line

File: app/Models/Comment.php [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+     * @var [type]
+     */
+    protected $fillable = [
+        'user_id',
+        'parent_id',
+        'body',
+        'created_at',
+        'updated_at',
+        'deleted_at'
+    ];
```
Comment (framgiaciv3) [inline app/Models/Comment.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Comment.php</br>Comma required after last value in array declaration

File: app/Models/Comment.php [hunk 32]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Get users: One to many
+     * @return [type] [description]
+     */
+    public function users() {
+        return $this->belongsTo('App\Models\User', 'user_id');
```
Comment (framgiaciv3) [inline app/Models/Comment.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Models/Comment.php</br>Opening brace should be on a new line

File: app/Models/Permission.php [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+class Permission extends Model
+{
+    use SoftDeletes;
+
+    /**
+     * [$table description]
+     * @var string
+     */
+    protected $table = "permissions";
+
```
Comment (framgiaciv3) [inline app/Models/Permission.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Models/Permission.php</br>String "permissions" does not require double quotes; use single quotes instead

File: app/Models/Like.php [hunk 34]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class Like extends Model
+{
+    use SoftDeletes;
```
Comment (BunnyPi04) [inline app/Models/Like.php]: Like có cần thiết phải dùng soft delete ko?

File: database/migrations/2018_11_05_021056_add_column_user_id_on_topics_table.php [hunk 35]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('topics', function (Blueprint $table) {
+            //
```
Comment (BunnyPi04) [inline database/migrations/2018_11_05_021056_add_column_user_id_on_topics_table.php]: sao lại để trống function down() vậy?

File: app/Models/Like.php (hunk 34 above)
Comment (NguyenCongViet) [reply app/Models/Like.php]: cũng không cần lắm chị ạ

File: database/migrations/2018_11_05_021056_add_column_user_id_on_topics_table.php (hunk 35 above)
Comment (NguyenCongViet) [reply database/migrations/2018_11_05_021056_add_column_user_id_on_topics_table.php]: lúc đầu e viết thiếu nên e bổ sung thêm cột user_id, nên e nghĩ không cần ghi vào hàm down()
vì e chạy migration có --table=topics rồi chị ạ

**Pull request merged**